dfastmi/kernel/BedLevelCalculator.py

```diff
@@ -0,0 +1,144 @@
+    """
+
+    def __init__(self, number_of_periods: int):
+        """
+        number_of_periods : int
+         Amount of the equilibrium bed level change for each respective discharge period available.
+        """
+        self.number_of_periods = number_of_periods
+
+    def get_element_wise_maximum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise maximum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmax : numpy.ndarray
+            Maximum bed level change.
+        """
+        return numpy.maximum.reduce(dzb)
+
+    def get_element_wise_minimum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise minimum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmin : numpy.ndarray
+            minimum bed level change.
+        """
+        return numpy.minimum.reduce(dzb)
+
+    def get_linear_average(self, number_of_days: Vector, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the linear average from the given bed levels.
+
+        Arguments
+        ---------
+        number_of_days : Vector
+            A vector of periods indicating the number of days during which each discharge applies.
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzgem : numpy.ndarray
+            Yearly mean bed level change.
+        """
+        dzgem = 0
+        for i in range(self.number_of_periods):
+            dzgem += dzb[i] * (number_of_days[i] + number_of_days[i-1]) / 2
+        return dzgem
+
+    def get_bed_level_changes(self, dzq: List[numpy.ndarray], rsigma: Vector) -> List[numpy.ndarray]:
+        """
+        This routine computes the bed level changes.
+        This routine requires that dzq and rsigma have the same length.
+        A stagnant period can be represented by a period with rsigma = 1.
+
+        Arguments
+        ---------
+        dzq : List[numpy.ndarray]
+            A list of arrays containing the equilibrium bed level change for each respective discharge period.
+        rsigma : Vector
+            A tuple of relaxation factors, one for each period.
+
+        Returns
+        -------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+        """
+        vsigma = self.__compute_vsigma__(rsigma, dzq)
```

Comment:
I think you should not end the method name with __. Now it suggests this is a 'dunder' method, while by the looks of it, it is just a 'private' method.

Reply:
Same for the other methods ending in __ ofcourse ^^

Reply:
Ah I though __name__ signified that it is a private method, this is not the case?
A private method would be: __name?

Reply:
Discussed, @MRVermeulenDeltares will adjust

Reply:
done